File: solutions_5648941810974720_1/java/NoS/GCJ_1B_A.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class GCJ_1B_A {
		
	class TestCase {
		String S;
	}
	
	public TestCase readTestCase() throws Exception {
		TestCase tc = new TestCase();
		tc.S = nextLine();
		return tc;
	}
	
	
	String[] helper = {"ZERO","ONE","TWO","THREE","FOUR","FIVE","SIX","SEVEN","EIGHT","NINE"};
	
	public Map<Character, List<Integer>> prepareMap() {
		Map<Character, List<Integer>> m = new HashMap<Character, List<Integer>>();
		for (char c = 'A'; c <= 'Z'; c++) {
			for (int i = 0; i < helper.length; i++) {
				String s = helper[i];
				if (s.contains("" + c)) {
					List<Integer> l = m.get(c);
					if (l == null) {
						m.put(c, l = new ArrayList<Integer>());
					}
					l.add(i);
				}
			}
		}
		return m;
	}
	
	
	private int[] toIndexArray(String s) {
		int[] result = new int[26];
		for (int i = 0; i < s.length(); i++) {
			char c = s.charAt(i);
			result[c - 'A']++;
		}
		return result;
	}
	
	public void writeTestResult(TestCase tc) throws Exception {
		List<Integer> result = new ArrayList<Integer>();
		Map<Character, List<Integer>> m = prepareMap();
		int unresolvedChars = tc.S.length();
		int[] indexArray = toIndexArray(tc.S);
		while (unresolvedChars > 0) {
			for (Map.Entry<Character, List<Integer>> e : m.entrySet()) {
				if (e.getValue().size() != 1) {
					continue;
				}
				Integer phoneDigit = e.getValue().get(0);
				int charCount = indexArray[e.getKey().charValue() - 'A'];
				for (int cc = 0; cc < charCount; cc++) {
					result.add(phoneDigit);
					String stringDigit = helper[phoneDigit];
					for (int i = 0; i < stringDigit.length(); i++) {
						char letter = stringDigit.charAt(i);
						indexArray[letter - 'A'] -= 1;
						unresolvedChars -= 1;
					}
				}
				//Update resolve map
				for (Map.Entry<Character, List<Integer>> ne : m.entrySet()) {
					ne.getValue().remove(phoneDigit);
				}
			}
		}
		Collections.sort(result);
		String phoneNumber = result.toString().replaceAll("[^0-9]*", "");
		System.out.println(phoneNumber);
		bw.write(phoneNumber);
	}

	//=======================================================
	//=======================================================
	//=======================================================

	static final String path = ".";
	BufferedReader br;
	BufferedWriter bw;
	
	String currentLine;
	String[] currentPieces;
	int currentIndex = 0;
	
	String nextLine() throws IOException {
		currentLine = br.readLine();
		currentIndex = 0;
		currentPieces = currentLine.split(" ");
		return currentLine;
	}
	
	String[] pieces() {
		return currentPieces;
	}
	
	String nextString() {
		return currentPieces[currentIndex++];
	}
	
	Long nextLong() {
		return Long.valueOf(currentPieces[currentIndex++]);
	}
	
	Double nextDouble() {
		return Double.valueOf(currentPieces[currentIndex++]);
	}
	
	
	public void solve() throws Exception {
		br = new BufferedReader(new FileReader(path+"/input"));
		bw = new BufferedWriter(new FileWriter(path+"/output"));
		int testCases = Integer.valueOf(br.readLine()).intValue();
		for (int i = 0; i < testCases; i++) {
			System.out.print("Test case " + (i + 1) + ": ");
			long time = System.nanoTime();
			TestCase testCase = readTestCase();
			System.out.print(" read in " + (System.nanoTime() - time) / 1E3 + " ms, ");
			time = System.nanoTime();
			bw.write("Case #" + (i + 1) + ": ");
			writeTestResult(testCase);
			bw.write("\n");
			System.out.println("solved in "+(System.nanoTime()-time)/1E3+" ms");
		}
		bw.flush();
		bw.close();
		br.close();
	}
	
	public static void main(String[] args) throws Exception  {
		GCJ_1B_A jam = new GCJ_1B_A();
		jam.solve();
	}
}

File: solutions_5648941810974720_1/C++/FrozenWinters/gcj1bA.cc
#include <fstream>
#include <sstream>

using namespace std;

int main(int argc, char* argv[]){
	ifstream in(argv[1]);
	ofstream out("phone.out");
  int T;
  in >> T;
  for(int C=1; C<=T; C++){
    string S;
    in >> S;
    int letters[26] = {};
    int numbers[10] = {};
    for(int i=0; i<S.length(); i++){
      letters[S[i]-'A']++;
    }
    //ZERO SIX SEVEN FOUR TWO FIVE EIGHT THREE ONE  NINE
    int n = letters['X'-'A'];
    numbers[6] = n;
    letters['S'-'A'] -= n;
    letters['I'-'A'] -= n;
    letters['X'-'A'] -= n;
    n = letters['Z'-'A'];
    numbers[0] = n;
    letters['Z'-'A'] -= n;
    letters['E'-'A'] -= n;
    letters['R'-'A'] -= n;
    letters['O'-'A'] -= n;
    n = letters['S'-'A'];
    numbers[7] = n;
    letters['S'-'A'] -= n;
    letters['E'-'A'] -= n;
    letters['V'-'A'] -= n;
    letters['E'-'A'] -= n;
    letters['N'-'A'] -= n;
    n = letters['U'-'A'];
    numbers[4] = n;
    letters['F'-'A'] -= n;
    letters['O'-'A'] -= n;
    letters['U'-'A'] -= n;
    letters['R'-'A'] -= n;
    n = letters['W'-'A'];
    numbers[2] = n;
    letters['T'-'A'] -= n;
    letters['W'-'A'] -= n;
    letters['O'-'A'] -= n;
    n = letters['V'-'A'];
    numbers[5] = n;
    letters['F'-'A'] -= n;
    letters['I'-'A'] -= n;
    letters['V'-'A'] -= n;
    letters['E'-'A'] -= n;
    n = letters['G'-'A'];
    numbers[8] = n;
    letters['E'-'A'] -= n;
    letters['I'-'A'] -= n;
    letters['G'-'A'] -= n;
    letters['H'-'A'] -= n;
    letters['T'-'A'] -= n;
    n = letters['R'-'A'];
    numbers[3] = n;
    letters['T'-'A'] -= n;
    letters['H'-'A'] -= n;
    letters['R'-'A'] -= n;
    letters['E'-'A'] -= n;
    letters['E'-'A'] -= n;
    n = letters['O'-'A'];
    numbers[1] = n;
    letters['O'-'A'] -= n;
    letters['N'-'A'] -= n;
    letters['E'-'A'] -= n;
    n = letters['I'-'A'];
    numbers[9] = n;
    letters['N'-'A'] -= n;
    letters['I'-'A'] -= n;
    letters['N'-'A'] -= n;
    letters['E'-'A'] -= n;
    stringstream ss;
    for(int j=0; j<10; j++){
      for(int k=0; k<numbers[j]; k++){
        ss << j;
      }
    }
    out << "Case #" << C << ": " << ss.str() << endl;
  }
}
